File: editorial/chap05/answer_B34.py
# 入力
N, X, Y = map(int, input().split())
A = list(map(int, input().split()))

# Grundy 数を計算
XOR_Sum = 0
for i in range(N):
	Grundy = [0, 0, 1, 1, 2] # 5 で割った余りに対応する Grundy 数
	XOR_Sum ^= Grundy[A[i] % 5]

# 出力
if XOR_Sum != 0:
	print("First")
else:
	print("Second")

File: editorial/chap05/cpp/answer_B34.cpp
#include <iostream>
using namespace std;

long long N, X, Y, A[100009];

int main() {
	// 入力
	cin >> N >> X >> Y;
	for (int i = 1; i <= N; i++) cin >> A[i];

	// Grundy 数を計算
	int XOR_Sum = 0;
	for (int i = 1; i <= N; i++) {
		if (A[i] % 5 == 0 || A[i] % 5 == 1) XOR_Sum ^= 0;
		if (A[i] % 5 == 2 || A[i] % 5 == 3) XOR_Sum ^= 1;
		if (A[i] % 5 == 4) XOR_Sum ^= 2;
	}

	// 出力
	if (XOR_Sum != 0) cout << "First" << endl;
	if (XOR_Sum == 0) cout << "Second" << endl;
	return 0;
}
